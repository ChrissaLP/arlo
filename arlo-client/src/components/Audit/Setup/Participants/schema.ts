import * as Yup from 'yup'

const participantsSchema = Yup.object().shape({
  state: Yup.string().required('Required'),
})

export default participantsSchema
