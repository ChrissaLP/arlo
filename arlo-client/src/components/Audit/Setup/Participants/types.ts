export interface IValues {
  state: string
}
