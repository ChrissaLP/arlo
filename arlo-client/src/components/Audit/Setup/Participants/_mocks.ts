const jurisdictionFile = new File(
  [
    'Jurisdiction","Admin Email"',
    '"Death Star","wtarkin@empire.gov"',
    '"Hoth","admin@rebelalliance.ninja"',
    '"Tatooine","jabba@hutt.biz"',
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
