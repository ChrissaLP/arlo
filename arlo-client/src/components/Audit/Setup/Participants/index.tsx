/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react'
import { Formik, FormikProps, Form, Field, ErrorMessage } from 'formik'
import styled from 'styled-components'
import { HTMLSelect, Spinner } from '@blueprintjs/core'
import { IAudit } from '../../../../types'
import FormWrapper from '../../../Form/FormWrapper'
import FormButtonBar from '../../../Form/FormButtonBar'
import FormButton from '../../../Form/FormButton'
import { IValues } from './types'
import states from './states'
import schema from './schema'
import { ErrorLabel } from '../../../Form/_helpers'

export const Select = styled(HTMLSelect)`
  margin-top: 5px;
`

const initialValues = {
  state: '',
}

interface IProps {
  audit: IAudit
  nextStage: () => void
}

const Participants: React.FC<IProps> = ({ audit, nextStage }: IProps) => {
  const [isLoading, setIsLoading] = useState(false)
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={schema}
      onSubmit={v => {
        setIsLoading(true)
        console.log(v)
        nextStage()
      }}
    >
      {({ handleSubmit, setFieldValue }: FormikProps<IValues>) => (
        <Form data-testid="form-one">
          <FormWrapper title="Participants">
            <label htmlFor="state">
              Choose your state from the options below
              <br />
              <Field
                component={Select}
                id="state"
                data-testid="state-field"
                name="state"
                onChange={(e: React.FormEvent<HTMLSelectElement>) =>
                  setFieldValue('state', e.currentTarget.value)
                }
                disabled={!!audit.frozenAt}
                options={states}
              />
            </label>
            <ErrorMessage name="state" component={ErrorLabel} />
          </FormWrapper>
          {isLoading && <Spinner />}
          {!audit.contests.length && !isLoading && (
            <FormButtonBar>
              <FormButton
                type="submit"
                intent="primary"
                disabled={!!audit.frozenAt}
                onClick={handleSubmit}
              >
                Submit &amp; Next
              </FormButton>
            </FormButtonBar>
          )}
        </Form>
      )}
    </Formik>
  )
}

export default Participants
