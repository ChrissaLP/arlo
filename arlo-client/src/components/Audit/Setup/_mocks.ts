import { ElementType } from '../../../types'
import { setupStages } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
) => {
  const prevTitle = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    title: prevTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextTitle = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    title: nextTitle,
    active: false,
    activate: jest.fn(),
    state,
  }
  return { prevStage, nextStage }
}

export default relativeStages
