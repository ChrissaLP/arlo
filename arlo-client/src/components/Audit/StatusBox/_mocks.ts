import {
  IJurisdictionsResponse,
  FileProcessingStatus,
} from './getJurisdictions'
import { IRound } from '../../../types'

export const jurisdictionMocks: {
  [key in
    | 'none'
    | 'noUploads'
    | 'halfUploads'
    | 'allFinishedFirstRound']: IJurisdictionsResponse
} = {
  none: { jurisdictions: [] },
  noUploads: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
    ],
  },
  halfUploads: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: null,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: null,
        ballotManifest: {
          file: null,
          numBallots: null,
          numBatches: null,
          processing: null,
        },
      },
    ],
  },
  allFinishedFirstRound: {
    jurisdictions: [
      {
        id: 'jurisdiction-1',
        name: 'Jurisdiction One',
        currentRoundStatus: 1,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
      {
        id: 'jurisdiction-2',
        name: 'Jurisdiction Two',
        currentRoundStatus: 1,
        ballotManifest: {
          file: 'ballot manifest',
          numBallots: 10,
          numBatches: 1,
          processing: FileProcessingStatus.Processed,
        },
      },
    ],
  },
}

export const roundMocks: {
  [key in
    | 'none'
    | 'firstRoundComplete'
    | 'firstRoundIncomplete'
    | 'secondRoundIncomplete']: { rounds: IRound[] }
} = {
  none: { rounds: [] },
  firstRoundIncomplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: null,
            sampleSizeOptions: null,
          },
        ],
        endedAt: null,
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
    ],
  },
  firstRoundComplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: true,
              pvalue: 0.00020431431380638307,
            },
            id: 'contest-1',
            results: {
              'choice-1': 100,
              'choice-2': 167,
            },
            sampleSize: 379,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: '2019-07-18T16:59:34.000Z',
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
    ],
  },
  secondRoundIncomplete: {
    rounds: [
      {
        contests: [
          {
            endMeasurements: {
              isComplete: false,
              pvalue: 1,
            },
            id: 'contest-1',
            results: {
              'choice-1': 0,
              'choice-2': 0,
            },
            sampleSize: null,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: '2019-07-18T16:35:07.000Z',
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-1',
      },
      {
        contests: [
          {
            endMeasurements: {
              isComplete: null,
              pvalue: null,
            },
            id: 'contest-1',
            results: {},
            sampleSize: null,
            sampleSizeOptions: [
              { size: 269, type: 'ASN', prob: null },
              { size: 379, prob: 0.8, type: null },
              { size: 78, prob: null, type: null },
            ],
          },
        ],
        endedAt: null,
        startedAt: '2019-07-18T16:34:07.000Z',
        id: 'round-2',
      },
    ],
  },
}
