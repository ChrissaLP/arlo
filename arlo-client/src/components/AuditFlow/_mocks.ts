import { IAuditBoard } from '../../types'

export const dummyBoard: IAuditBoard[] = [
  {
    id: '123',
    name: 'Audit Board #1',
    members: [],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
        id: '1',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
        id: '2',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
        id: '3',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'NOT_AUDITED',
        vote: null,
        comment: '',
        id: '4',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '5',
      },
    ],
  },
  {
    id: '123',
    name: 'Audit Board #1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    ballots: [
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '1',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '2',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '3',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '4',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '5',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '6',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '7',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '8',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '9',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '10',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '11',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '12',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '13',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '14',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '15',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '16',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '17',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '18',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '19',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '20',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '21',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '22',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '23',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '24',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '25',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '26',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '27',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '28',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '29',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '30',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '31',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '32',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '33',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '34',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '35',
      },
      {
        tabulator: '11',
        batch: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        position: '313',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '36',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '37',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '38',
      },
      {
        tabulator: '17',
        batch: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        position: '2112',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '39',
      },
      {
        tabulator: '23',
        batch: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        position: '1789',
        status: 'AUDITED',
        vote: null,
        comment: '',
        id: '40',
      },
    ],
  },
]

export default {
  dummyBoard,
}
