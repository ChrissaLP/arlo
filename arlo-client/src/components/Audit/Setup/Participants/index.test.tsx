import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { statusStates } from '../../_mocks'
import Participants from './index'

describe('Audit Setup > Contests', () => {
  it('renders empty state correctly', () => {
    const { container } = render(
      <Participants audit={statusStates[0]} nextStage={jest.fn()} />
    )
    expect(container).toMatchSnapshot()
  })

  it('selects a state and submits it', async () => {
    const nextStageMock = jest.fn()
    const { getByText, getByTestId } = render(
      <Participants audit={statusStates[0]} nextStage={nextStageMock} />
    )

    fireEvent.change(getByTestId('state-field'), {
      target: { value: 'Alabama' },
    })
    // await wait(() => expect(queryByText('Your state is: Alabama')).toBeTruthy())
    fireEvent.click(getByText('Submit & Next'), { bubbles: true })
    await wait(() => expect(nextStageMock).toHaveBeenCalledTimes(1))
  })
})
