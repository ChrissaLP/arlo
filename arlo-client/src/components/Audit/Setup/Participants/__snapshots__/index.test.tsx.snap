// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit Setup > Contests renders empty state correctly 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH hfheQC"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Participants
      </h2>
      <label
        for="state"
      >
        Choose your state from the options below
        <br />
        <div
          class="bp3-html-select sc-bZQynM imeWhx"
        >
          <select
            data-testid="state-field"
            field="[object Object]"
            form="[object Object]"
            id="state"
          >
            <option
              value=""
            />
            <option
              value="Alabama"
            >
              Alabama
            </option>
            <option
              value="Alaska"
            >
              Alaska
            </option>
            <option
              value="Arizona"
            >
              Arizona
            </option>
            <option
              value="Arkansas"
            >
              Arkansas
            </option>
            <option
              value="California"
            >
              California
            </option>
            <option
              value="Colorado"
            >
              Colorado
            </option>
            <option
              value="Connecticut"
            >
              Connecticut
            </option>
            <option
              value="Delaware"
            >
              Delaware
            </option>
            <option
              value="Florida"
            >
              Florida
            </option>
            <option
              value="Georgia"
            >
              Georgia
            </option>
            <option
              value="Hawaii"
            >
              Hawaii
            </option>
            <option
              value="Idaho"
            >
              Idaho
            </option>
            <option
              value="Illinois"
            >
              Illinois
            </option>
            <option
              value="Indiana"
            >
              Indiana
            </option>
            <option
              value="Iowa"
            >
              Iowa
            </option>
            <option
              value="Kansas"
            >
              Kansas
            </option>
            <option
              value="Kentucky"
            >
              Kentucky
            </option>
            <option
              value="Louisiana"
            >
              Louisiana
            </option>
            <option
              value="Maine"
            >
              Maine
            </option>
            <option
              value="Maryland"
            >
              Maryland
            </option>
            <option
              value="Massachusetts"
            >
              Massachusetts
            </option>
            <option
              value="Michigan"
            >
              Michigan
            </option>
            <option
              value="Minnesota"
            >
              Minnesota
            </option>
            <option
              value="Mississippi"
            >
              Mississippi
            </option>
            <option
              value="Missouri"
            >
              Missouri
            </option>
            <option
              value="Montana"
            >
              Montana
            </option>
            <option
              value="Nebraska"
            >
              Nebraska
            </option>
            <option
              value="Nevada"
            >
              Nevada
            </option>
            <option
              value="New Hampshire"
            >
              New Hampshire
            </option>
            <option
              value="New Jersey"
            >
              New Jersey
            </option>
            <option
              value="New Mexico"
            >
              New Mexico
            </option>
            <option
              value="New York"
            >
              New York
            </option>
            <option
              value="North Carolina"
            >
              North Carolina
            </option>
            <option
              value="North Dakota"
            >
              North Dakota
            </option>
            <option
              value="Ohio"
            >
              Ohio
            </option>
            <option
              value="Oklahoma"
            >
              Oklahoma
            </option>
            <option
              value="Oregon"
            >
              Oregon
            </option>
            <option
              value="Pennsylvania"
            >
              Pennsylvania
            </option>
            <option
              value="Rhode Island"
            >
              Rhode Island
            </option>
            <option
              value="South Carolina"
            >
              South Carolina
            </option>
            <option
              value="South Dakota"
            >
              South Dakota
            </option>
            <option
              value="Tennessee"
            >
              Tennessee
            </option>
            <option
              value="Texas"
            >
              Texas
            </option>
            <option
              value="Utah"
            >
              Utah
            </option>
            <option
              value="Vermont"
            >
              Vermont
            </option>
            <option
              value="Virginia"
            >
              Virginia
            </option>
            <option
              value="Washington"
            >
              Washington
            </option>
            <option
              value="West Virginia"
            >
              West Virginia
            </option>
            <option
              value="Wisconsin"
            >
              Wisconsin
            </option>
            <option
              value="Wyoming"
            >
              Wyoming
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </label>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button bp3-intent-primary sc-ifAKCX zdCNs"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          Submit & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;
